### PR TITLE
Update Redis code samples

### PR DESCRIPTION
**Pull request by xiangyan99**
Fixes https://github.com/Azure/azure-sdk-for-python/issues/28699

### CODE REVIEW

File: sdk/identity/azure-identity/samples/azure-aad-auth-with-redis-py.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ Integrate the logic in your application code to fetch an Azure AD access token v
 import redis
 from azure.identity import DefaultAzureCredential
 
-scope = "https://*.cacheinfra.windows.net:10225/appid/.default"  # The scope will be changed for AAD Public Preview
+scope = "acca5fbb-b7e4-4009-81f1-37e38fd66d78/.default"  # The scope will be changed for AAD Public Preview
```
Comment (g2vinay) [inline sdk/identity/azure-identity/samples/azure-aad-auth-with-redis-py.md:39]: The comment can be updated to "The current scope is for public preview and may change for GA release."

**Comment by g2vinay** [comment]
> 

Thanks for reminding about this discussion @xiangyan99.
Since, its not a best practice, having a gist will be good.

Can we link out to the gist from samples doc, and also mention that it isn't recommended/best-practice, but users looking to ensure that connection is proactively refreshed can follow this as a reference ?

@samsaha-ms FYI

**Comment by samsaha-ms** [comment]
> > @xiangyan99 is it possible to add a sample which proactively re-authenticates the connection 2-5 minutes before token expiry. For Java we've added it for Jedis and Lettuce.
> 
> As we discussed, in order to do it, we need to create a background thread which is not the best practice for Python users. We can create a gist to share but we don't want to put it in samples.

I am not a python expert so I can't comment on it. But wanted to know why is background thread not recommended in python? Also, we need to ensure redis connection object for python library is thread safe otherwise we need to either synchronize it or some exception handling to recreate connections on erroneous event.

**Comment by samsaha-ms** [comment]
> > @xiangyan99 is it possible to add a sample which proactively re-authenticates the connection 2-5 minutes before token expiry. For Java we've added it for Jedis and Lettuce.
> 
> As we discussed, in order to do it, we need to create a background thread which is not the best practice for Python users. We can create a gist to share but we don't want to put it in samples.

I am not a python expert so I can't comment on it. But wanted to know why is background thread not recommended in python? Also, we need to ensure redis connection object for python library is thread safe otherwise we need to either synchronize it or some exception handling to recreate connections on errorneous event.

File: sdk/identity/azure-identity/samples/azure-aad-auth-with-redis.py [hunk 2]
```diff
@@ -46,6 +47,12 @@ def re_authentication():
     max_retry = 3
     for index in range(max_retry):
         try:
+            if _need_refreshing(token):
```
Comment (g2vinay) [inline sdk/identity/azure-identity/samples/azure-aad-auth-with-redis.py:50]: For the async scheduled logic in the gist, 
we can leave a comment here for users looking to use it that way, can reach out to us via GitHub and we can provide them the guidance/pointers.

**Review by g2vinay — APPROVED** [review]

**Pull request merged**